x-pack/plugins/enterprise_search/public/applications/shared/use_local_storage/use_local_storage.ts

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useState, useEffect } from 'react';
+
+export const useLocalStorage = <T>(key: string, defaultValue: T) => {
```

Comment:
Oh man every time I see typescript generics I'm just like amazed. 😂 

I've made this comment a few times now but I'll still stand by it, I'm not a huge fan of single letter generics not giving me much context as to what exactly I'm dealing with. I have a slight preference to trying to give it a bit more of a descriptive name, like:

```
 export const useLocalStorage = <Value>(key: string, defaultValue: Value) => { 
```

Although... actually, I'm a little confused. Local storage [forces everything to a string](https://stackoverflow.com/questions/13442819/html5-local-storage-and-variable-types) - I know this is a generic component, but if we actually want to deal with complex types (numbers, objs, etc.) we'd have to greatly modify / extend the component to JSON.stringify/JSON.parse and so forth.

Just IMO, I think we should simplify this component to assume strings for all values and type it that way as well. That way we can also lose generic types. What do you think?

```suggestion
export const useLocalStorage = (key: string, defaultValue: string) => {
```

Reply:
oh wow I'm really tired today, I only just noticed your localStorage component already does use JSON stringify/parse 🤦‍♀️ 

disregard my second suggestion in that case

Reply:
>  I'm not a huge fan of single letter generics not giving me much context as to what exactly I'm dealing with

I'll make this change to be consistent with other code in our project.

Reply:
c5922054b8b436a32c4e3102cf4df6753cd72a9f

Reply:
Thanks a ton Jason!